main() panics on os.Args[1] when invoked with no arguments

## Summary

`main()` accesses `os.Args[1]` on line 30 of `ussher.go` *before* the length check on line 48. Invoking `ussher` with no arguments crashes with a Go runtime panic instead of producing the intended `usage: ussher <username>` message.

## Reproduction

```
$ ./ussher
panic: runtime error: index out of range [1] with length 1
```

(The clean `log.Fatal("usage: ussher <username>")` further down at line 49 is unreachable for the no-args case.)

## Why this matters specifically for `ussher`

This binary lives in the SSH authentication path. A panic — rather than a clean error — produces a Go stack trace on stderr, which is mildly informative for an operator running `./ussher` by hand, but for any caller that processes ussher's output (sshd parsing stdout, a probe/health-check, an `--version` script that misorders args), the trace is unexpected and the exit code is 2 (Go runtime panic) rather than 1 (clean fatal). It's also the kind of crash a reachability scan would flag as "potentially exploitable" at a glance, even though it isn't.

The `--version` short-circuit ironically suffers the same issue — bare `./ussher` can never print version info because the args check happens first and panics before the version branch can match.

## Approach

Move the length check to the very top of `main`, before any `os.Args[1]` access:

```go
func main() {
    if len(os.Args) < 2 {
        // Pre-init-log fatal: write to stderr (the default log destination).
        log.Fatal("usage: ussher <username>")
    }

    if os.Args[1] == "--version" {
        PrintVersion()
        return
    }

    // ... existing security checks, initLog, etc.
}
```

This is also a natural seam for the `parseArgs` extraction proposed in #13 — the new function would do the length and `--version` handling and return either a trigger to print version, a typed error, or a username string.

## Acceptance criteria

- [ ] `./ussher` (no args) prints `usage: ussher <username>` to stderr and exits non-zero, no Go panic.
- [ ] `./ussher --version` continues to work.
- [ ] `./ussher root` (or any valid user) continues to work.
- [ ] A test case in `ussher_test.go` (per #13) covers the no-args path.

## Files

- `/home/user/ussher/ussher.go`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- `ussher` invoked with no arguments now prints `usage: ussher <username>`
+  and exits cleanly instead of panicking with a Go runtime "index out of
+  range" trace. The argument-count check used to live below the
+  `os.Args[1] == "--version"` access, making the bare-invocation case
+  unreachable. ([#29])
+
 ## [1.1.0] - 2026-04-27
 
 ### Added
@@ -93,4 +101,5 @@ Initial release.
 [#10]: https://github.com/dolph/ussher/pull/10
 [#12]: https://github.com/dolph/ussher/issues/12
 [#25]: https://github.com/dolph/ussher/pull/25
+[#29]: https://github.com/dolph/ussher/issues/29
 [#31]: https://github.com/dolph/ussher/issues/31

--- a/ussher.go
+++ b/ussher.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"errors"
 	"log"
 	"os"
 )
@@ -24,10 +25,26 @@ func initLog() {
 	log.SetOutput(file)
 }
 
+// validateArgs requires args to look like `ussher <single-arg>` (a username
+// or `--version`) and returns the single arg or a usage error. The wrong-arg-
+// count check has to live before any os.Args[1] access; otherwise invoking
+// `ussher` with no args panics with "index out of range".
+func validateArgs(args []string) (string, error) {
+	if len(args) != 2 {
+		return "", errors.New("usage: ussher <username>")
+	}
+	return args[1], nil
+}
+
 func main() {
+	arg, err := validateArgs(os.Args)
+	if err != nil {
+		log.Fatal(err)
+	}
+
 	// Support `ussher --version` to print versioning info about the executable
 	// before proceeding to security-hardening checks.
-	if os.Args[1] == "--version" {
+	if arg == "--version" {
 		PrintVersion()
 		return
 	}
@@ -44,13 +61,8 @@ func main() {
 	// a non-root user
 	initLog()
 
-	// Check if the input username is provided
-	if len(os.Args) != 2 {
-		log.Fatal("usage: ussher <username>")
-	}
-
 	// Check if the input username is valid
-	username := os.Args[1]
+	username := arg
 	if !isValidUser(username) {
 		log.Fatal("User not found")
 	}

--- a/ussher_test.go
+++ b/ussher_test.go
@@ -1,0 +1,30 @@
+package main
+
+import "testing"
+
+func TestValidateArgs(t *testing.T) {
+	cases := []struct {
+		name    string
+		args    []string
+		want    string
+		wantErr bool
+	}{
+		{"no args (just program name)", []string{"ussher"}, "", true},
+		{"empty args slice", []string{}, "", true},
+		{"single arg looks like username", []string{"ussher", "alice"}, "alice", false},
+		{"single arg is --version", []string{"ussher", "--version"}, "--version", false},
+		{"too many args", []string{"ussher", "alice", "extra"}, "", true},
+		{"too many args with --version", []string{"ussher", "--version", "extra"}, "", true},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := validateArgs(tc.args)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("validateArgs(%v) err = %v, wantErr %v", tc.args, err, tc.wantErr)
+			}
+			if got != tc.want {
+				t.Errorf("validateArgs(%v) = %q, want %q", tc.args, got, tc.want)
+			}
+		})
+	}
+}

